storage/streams/src/main/java/io/apicurio/registry/streams/diservice/DistributedService.java

```diff
@@ -104,7 +104,7 @@ public void close() {
 
     protected final S serviceForKey(K key) {
         StreamsMetadata smeta = streams.metadataForKey(storeName, key, keySerde.serializer());
-        if (smeta == StreamsMetadata.NOT_AVAILABLE) {
+        if (smeta == null || smeta == StreamsMetadata.NOT_AVAILABLE) {
```

Comment:
`null` really indicates that there is no matching metadata for the given store and key. _Maybe_ that can be a transient condition (to be honest I don't know, but I'm guessing the store might eventually have that key), or it could indicate a non-transient programming error in your streams application (i.e. you're asking for the wrong key, or the wrong store and it will never become non-null). In any case I think it would be useful to have a distinct exception message, since the `NOT_AVAILABLE` case is definitely expected to be transient.

Reply:
OK, will fix this ...

Reply:
FWIW I opened https://issues.apache.org/jira/browse/KAFKA-9663